src/pybind/mgr/mgr_module.py

```diff
@@ -379,6 +379,10 @@ def _load_func_metadata(cls: Any, f: HandlerFuncType) -> Tuple[str, Dict[str, An
         positional = True
         for index, arg in enumerate(full_argspec.args):
             if arg in cls.KNOWN_ARGS:
+                # record that this function takes an inbuf if it is present
+                # in the full_argspec and not already in the arg_spec
+                if arg == 'inbuf' and 'inbuf' not in arg_spec:
+                    arg_spec['inbuf'] = 'str'
                 continue
```

Comment:
Obviously it's not your code so you might not know, but any thoughts on why for the `KNOWN_ARGS` it wasn't appending them to the args list generally? It seems like the fact that it doesn't track if the python function actually supports these special args or not is sort of where the issue you're addressing comes from.

Reply:
It seems that KNOWN_ARGS is a list of things that are actual args (inbuf for example) and things that are placeholders (self, _) and stuff that's not args at all (like "return"). So it's a bit of a misnomer. It probably should have been called "STUFF_TO_IGNORE" ;-)  This issue is that the api in use (getfullargspec) is kinda crappy.
Since it makes no sense to add most of these to the args list `inbuf` was ignored like the rest, even though it *is* an argument, just a special cased one.